pkg/watcher/internal/metricsprovider/datadog_test.go

```diff
@@ -30,51 +23,258 @@ func TestNewDatadogClient(t *testing.T) {
+                    "unit": null
+                }
+            ],
+            "times": [
+                1724967300000,
+                1724967360000,
+                1724967420000,
+                1724967480000,
+                1724967540000,
+                1724967600000,
+                1724967660000,
+                1724967720000,
+                1724967780000,
+                1724967840000,
+                1724967900000,
+                1724967960000,
+                1724968020000,
+                1724968080000,
+                1724968140000,
+                1724968200000
+            ],
+            "values": [
+                [
+                    7.332664,
+                    10.399366,
+                    11.780529,
+                    10.082532,
+                    8.429297,
+                    7.017103,
+                    12.490895,
+                    7.128327,
+                    7.08206,
+                    5.555416,
+                    9.067136,
+                    9.532126,
+                    11.440966,
+                    10.396935,
+                    8.71661,
+                    6.193735
+                ],
+                [
+                    50.949683,
+                    50.950399,
+                    50.946066,
+                    50.950076,
+                    50.934947,
+                    50.941883,
+                    50.968234,
+                    50.984515,
+                    51.007935,
+                    51.011688,
+                    51.01305,
+                    51.022667,
+                    51.038384,
+                    51.044218,
+                    51.04624,
+                    51.04946
+                ],
+				[
+                    6.332664,
+                    11.399366,
+                    12.780529,
+                    13.082532,
+                    18.429297,
+                    17.017103,
+                    22.490895,
+                    17.128327,
+                    17.08206,
+                    15.555416,
+                    19.067136,
+                    19.532126,
+                    21.440966,
+                    20.396935,
+                    18.71661,
+                    16.193735
+                ],
+                [
+                    40.949683,
+                    40.950399,
+                    40.946066,
+                    40.950076,
+                    40.934947,
+                    40.941883,
+                    40.968234,
+                    40.984515,
+                    41.007935,
+                    41.011688,
+                    41.01305,
+                    41.022667,
+                    41.038384,
+                    41.044218,
+                    41.04624,
+                    41.04946
+                ]
+            ]
+        }
+    }
+}`
 
-	client, err := NewDatadogClient(opts)
+	bytes := []byte(metricData)
+	resp := datadogV2.NewTimeseriesFormulaQueryResponseWithDefaults()
```

Comment:
it seems we are doing unit tests for datadog API not loadwatcher api.
The idea of unit tests are to make sure our APIs are good.

Reply:
Unfortunately, it may not be possible to fake the response from Datadog api client. This way, at least we could test the parsing logic.